core/prisma/migrations/20241107193617_add_capabilities_to_db/migration.sql

```diff
@@ -0,0 +1,147 @@
+-- CreateEnum
+CREATE TYPE "Capabilities" AS ENUM (
+    'movePub',
+    'createPub',
+    'viewPub',
+    'deletePub',
+    'updatePubValues',
+    'createRelatedPub',
+    'createPubWithForm',
+    'editPubWithForm',
+    'createPubField',
+    'archivePubField',
+    'editPubField',
+    'createPubType',
+    'editPubType',
+    'deletePubType',
+    'runAction',
+    'viewStage',
+    'createStage',
+    'manageStage',
+    'deleteStage',
+    'addPubMember',
+    'removePubMember',
+    'addStageMember',
+    'removeStageMember',
+    'addFormMember',
+    'removeFormMember',
+    'addCommunityMember',
+    'removeCommunityMember',
+    'manageMemberGroups',
+    'addCommunity',
+    'editCommunity',
+    'createForm',
+    'editForm',
+    'archiveForm',
+    'createApiToken',
+    'revokeApiToken'
+);
+
+-- CreateEnum
+CREATE TYPE "MembershipType" AS ENUM ('community', 'stage', 'pub', 'form');
+
+-- CreateTable
+CREATE TABLE "membership_capabilities" (
+    "role" "MemberRole" NOT NULL,
+    "type" "MembershipType" NOT NULL,
+    "capability" "Capabilities" NOT NULL,
+
+    CONSTRAINT "membership_capabilities_pkey" PRIMARY KEY ("role","type","capability")
+);
+
+INSERT INTO "membership_capabilities"
+VALUES
```

Comment:
@gabestein the table below maps role and membership type to capabilities for stages, pubs, and communities. Could you review to make sure there's no missing or extra permissions for various roles? I also think I've missed a few required capabilities, which you're welcome to note here or we can just add them along with the features that require them.

Reply:
Thanks! I only see a few things:

- Pub Contributors should be able to create related pubs, per the "old" tab in #594 .
- Pub Editors should be able to move and take actions, per the "old" tab in #594 .
- Pub Editors should be able to see all related pubs, per the "old" tab in #594 (not sure if that'll be handled here?)
- I'm not positive Stage Contributors should be able to `editPubWithForm` if we're going to implement form permissions separately?
- Stage Editors should be able to see all pubs related to pubs in the stage, per the "old" tab in #638 (not sure if that'll be handled here)

Reply:
great, thanks for checking. i'll make these updates alongside my next PR